lib/Lower/Bridge.cpp

```diff
@@ -160,7 +158,7 @@ class FirConverter : public Fortran::lower::AbstractConverter {
 
   mlir::Location getCurrentLocation() override final { return toLocation(); }
   mlir::Location genLocation() override final {
-    return builder->getUnknownLoc();
+    return mlir::UnknownLoc::get(&mlirContext);
```

Comment:
This should use the builder, not a direct call to `get`.

Reply:
Is there a downside of directly using `get` ? The issue of using the builder is that it does not always exists when this is called (I discovered it the hard way....), and I find it hard to guarantee that it would.
The place where I found the issue with the added tests is:
```
    if (!func)
      func = createNewFunction(name, funit.symbol); 
    builder = new Fortran::lower::FirOpBuilder(func);
```
`createNewFunction` calls `toLocation()` that calls `genLocation()`.

Reply:
OK, now I am confused. Is `genLocation()` always just the Unknown location? That doesn't seem right for creating a function. Most functions will definitely have a source location, non?

Reply:
It was an issue with anonymous program for which we do not have the location for now. I saw your reworked this a bit in https://github.com/jeanPerier/f18/commit/5052fcb186e5fede7acb72288adfcce67ac3cc3b
There is a bit of redundancy with `extractLocation` from my change, I removed `extractLocation`.

Reply:
Yes. I tried the thread the location for the function itself.